File: C#/CES.cs
using System;
using System.Collections.Generic;
using System.Linq;
using MathNet.Numerics.LinearAlgebra;
using MathNet.Numerics.LinearAlgebra.Double;

namespace CES
{
    class Program
    {
        static void Main(string[] args)
        {
            // -2 7 -9 0 -10 -5
            Console.WriteLine("Enter your three coordinate points, separated by spaces.");
            string[] input = Console.ReadLine().Split(' ');
            List<double> r=input.Select(x=>double.Parse(x)).ToList();
            double a=r[0],b=r[1],c=r[2],d=r[3],e=r[4],f=r[5],D,E,F;
            Matrix<double> m = DenseMatrix.OfArray(new double[,] {{-a*a-b*b},{-c*c-d*d},{-e*e-f*f}});
            Matrix<double> m2 = DenseMatrix.OfArray(new double[,]{{a,b,1},{c,d,1},{e,f,1}});
            Matrix<double> DEF = m2.Solve(m);
            D = DEF[0,0]/-2;
            E = DEF[1,0]/-2;
            F = DEF[2,0];
            Console.WriteLine("(x - "+D+")\u00b2 + (y - "+E+")\u00b2 = "+(-F+D*D+E*E).ToString());
        }
    }
}

File: Java/CES.java
import java.util.Scanner;
public class CES {
    private static double avg(double[] a, int b, int c) {return (a[b]+a[c])/2;}
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Enter your three coordinate points, separated by spaces:");
        double[] c = new double[6];
        for (int i = 0; i < 6; i++) c[i] = scan.nextDouble();
        double x0 = avg(c,0,2),y0 = avg(c,1,3),x1 = avg(c,0,4),y1 = avg(c,1,5),
        s = (c[0]-c[2])/(c[3]-c[1]), s2 = (c[0]-c[4])/(c[5]-c[1]), h, k, r2;
        h = (y1-y0+s*x0-s2*x1) / (s-s2);
	k = (h-x0)*s+y0;
	r2 = (h-c[0])*(h-c[0])+(k-c[1])*(k-c[1]);
	System.out.printf("(x - %.3f)² + (y - %.3f)² = %.3f",h,k,r2);
    }
}
